[WIP] Run commands via issue comments

rudehn opened the pull request:
Start working towards #192 by retrieving issue comments

So what's going on here? Good question!

This provides the ability to run commands to chaosbot through issue comments. Current functionality enables closing and (maybe?) opening of issues, however this can be expanded to be as flexible as needed.

This could definitely use some review and testing from other people.

benbarth committed:
Code cleanup and removed unused imports

benbarth committed:
Update README.md

Add ChaosBot badge

benbarth committed:
Revert "Code cleanup and removed unused imports"

This reverts commit b4fc4f3d767c7f738dcb5cd784826dcc02c577bb.

benbarth committed:
Remove unused import

xyproto committed:
Only insignificant nitpicks

xyproto committed:
Take it easy

flibustier committed:
Add Windows support for Docker

xyproto committed:
Remove obsolete Dockerfile

phil-r committed:
Add homepage to the repo
See github api docs here: https://developer.github.com/v3/repos/#edit

tarunbatra committed:
Fixes repo url parsing

The logic was resulting in `<username>/c` instead of `username/chaos`.

Corrected it.

xyproto committed:
Update settings.py

rudehn committed:
Get issue comments

Start working towards #192 by retrieving issue comments

Commit:
Website Tweaks

— Changed the “Chaos” text color to red (more suiting).
— Added a notification telling the user to hover over the logo, it is
hard to tell it is animated otherwise.
— Added a moving background that adds more to the “chaos”. (based on
Canva Moving Background from Codepen and it is a background I also use
for my own website.)

rudehn committed:
Create poll_read_issue_comments

rudehn committed:
Update __init__.py

rudehn committed:
Update settings.py

chaosbot committed:
merging PR #274: Code cleanup and removed unused imports

https://github.com/chaosbot/Chaos/pull/274: Code cleanup and removed unused imports

Description:
Minor fixes and removing unused imports

:ok_woman: PR passed with a vote of 10 for and 0 against, with a weighted total of 9.0 and a threshold of 6.1.

Vote record:
@DasSkelett: 1
@G4Zz0L1: 1
@Guard13007: 1
@Lastrik: 1
@Lomeli12: 1
@andrewda: 1
@benbarth: 1
@pchauncey: 1
@shnist: 1
@tversteeg: 1

chaosbot committed:
merging PR #277: Add Windows support for Docker

https://github.com/chaosbot/Chaos/pull/277: Add Windows support for Docker

Description:
The concept of developing under Windows may seem strange, but we shouldn't put any developers aside, despite our differences!

:ok_woman: PR passed with a vote of 14 for and 0 against, with a weighted total of 14.0 and a threshold of 6.1.

Vote record:
@J0hn-: 1
@MUCHZER: 1
@Moussto: 1
@N00D13: 1
@SylvainThrd: 1
@amoffat: 1
@cjoh: 1
@costin-zaharia: 1
@hongaar: 1
@jeroeness: 1
@pchauncey: 1
@rhengles: 1
@rudehn: 1
@tarunbatra: 1

rudehn committed:
Update and rename poll_read_issue_comments to poll_read_issue_comments.py

rudehn committed:
covert to dict

rudehn committed:
Update comments.py

marcosrjjunior committed:
Adding github corner at website

o3LL committed:
fix indent

rudehn committed:
Add command structure

Not really feeling like implementing all the commands myself. Perhaps somebody else might take a look?

rudehn committed:
Create issues.py

rudehn committed:
integrate close/open issue

rudehn committed:
bug fix

rudehn committed:
add voting to commands

rudehn committed:
Update settings.py

rudehn committed:
bug fix

o3LL committed:
refactor using no jquery

o3LL committed:
removed jquery import

o3LL committed:
fixed indent

o3LL committed:
moved fa import

o3LL committed:
little style update

tarunbatra committed:
Removes .git if present in remote url

rudehn committed:
add issue comment last update time

rudehn committed:
add voting window for issue comment commands

rudehn committed:
Update settings.py

o3LL committed:
added notification to move hover the canvas

o3LL committed:
refactor event

o3LL committed:
reindent + removed console.log

rudehn committed:
bug fix

o3LL committed:
removed console log and unbroke vwidth

o3LL committed:
wow smootb anim such fade

chaosbot committed:
merging PR #203: Only insignificant nitpicks

https://github.com/chaosbot/Chaos/pull/203: Only insignificant nitpicks

Description:
* Add files to `.gitignore`
* Run `autopep8` on all the Python scripts
* Add UTF-8 declarations to the Python scripts
* Spellcheck all the documentation
* Place dangling Python code in functions
* Use `#!/bin/bash` instead of `#!/usr/bin/env bash`
* Quote shell variables that may contain filenames or directory names with spaces or dashes
* Whitespace changes
* Move the `Dockerfile` to its own directory


:ok_woman: PR passed with a vote of 17 for and 2 against, with a weighted total of 15.0 and a threshold of 6.1.

Vote record:
@BenJenkinson: 1
@Dentosal: 1
@ECrownofFire: 1
@EtherTyper: 1
@Vorlent: -1
@albinekb: 1
@amoffat: 1
@andrewda: 1
@dcstone09: 1
@eukaryote31: 1
@evysan: 1
@hongaar: 1
@istarkov: 1
@jon1467: 1
@mjosh954: 1
@molovo: -1
@ozyx: 1
@xyproto: 1
@zakrid: 1

chaosbot committed:
merging PR #275: Take it easy

https://github.com/chaosbot/Chaos/pull/275: Take it easy

Description:
I think the voting window is too short, especially for people in different timezones than the US.

After the initial interest in chaosbot is slowly starting to fade, it makes sense to slow down the time limits in the voting process just a bit.

:ok_woman: PR passed with a vote of 21 for and 1 against, with a weighted total of 20.0 and a threshold of 6.1.

Vote record:
@G4Zz0L1: 1
@GantMan: 1
@J0hn-: 1
@Jezza: 1
@N00D13: 1
@SylvainThrd: 1
@anythingbot: 1
@elamperti: 1
@hongaar: 1
@jeroeness: 1
@jesshillier: 1
@magarcia: 1
@max-wittig: 1
@nstanard: 1
@phil-r: 1
@rhengles: -1
@rudehn: 1
@shnist: 1
@valfirst: 1
@viktorsec: 1
@vivekzhere: 1
@xyproto: 1

rudehn committed:
save previously ran commands

chaosbot committed:
merging PR #279: Fixes repo url parsing

https://github.com/chaosbot/Chaos/pull/279: Fixes repo url parsing

Description:
The logic was resulting in `<username>/c` instead of `username/chaos`.

Corrected it.

:ok_woman: PR passed with a vote of 11 for and 0 against, with a weighted total of 11.0 and a threshold of 6.2.

Vote record:
@BenMorganIO: 1
@J0hn-: 1
@PlasmaPower: 1
@amoffat: 1
@datitisev: 1
@hongaar: 1
@hubert-pomorski: 1
@reddraggone9: 1
@rhengles: 1
@tarunbatra: 1
@vitorduque: 1

flibustier committed:
Dynamic Voting Window

chaosbot committed:
Revert "merging PR #271: get true PR pushed_at datetime"

This reverts commit ef69d7036d1b5eccf2997adc4a9dfd9557a7f4a3, reversing
changes made to c2498c4621a5f890bd704d568a9192acd9ab5711.

this is a manual revert due to the nature of the bug it causes (making it nearly
impossible for PRs to get merged in)

chaosbot committed:
merging PR #287: Adding github corner at website

https://github.com/chaosbot/Chaos/pull/287: Adding github corner at website

Description:
![gcorner](https://cloud.githubusercontent.com/assets/5287262/26499659/d0319444-4209-11e7-8f55-9204ad916e68.PNG)


:ok_woman: PR passed with a vote of 13 for and 0 against, with a weighted total of 13.0 and a threshold of 6.2.

Vote record:
@BenMorganIO: 1
@Leigende: 1
@MUCHZER: 1
@N00D13: 1
@abrad1212: 1
@albinekb: 1
@amoffat: 1
@magarcia: 1
@marcosrjjunior: 1
@meltahawy: 1
@nstanard: 1
@redg3ar: 1
@rudehn: 1

chaosbot committed:
merging PR #280: Add ChaosBot badge to README.md

https://github.com/chaosbot/Chaos/pull/280: Add ChaosBot badge to README.md

Description:
Add ChaosBot badge

:ok_woman: PR passed with a vote of 12 for and 2 against, with a weighted total of 10.0 and a threshold of 6.2.

Vote record:
@J0hn-: 1
@MUCHZER: 1
@Reuh: 1
@Smittyvb: 1
@Xavi1321: 1
@abrad1212: 1
@albinekb: 1
@benbarth: 1
@datitisev: 1
@hongaar: 1
@redg3ar: -1
@rhengles: -1
@rudehn: 1
@valfirst: 1

chaosbot committed:
merging PR #278: Add homepage to the repo

https://github.com/chaosbot/Chaos/pull/278: Add homepage to the repo

Description:
See github api docs here: https://developer.github.com/v3/repos/#edit

:ok_woman: PR passed with a vote of 18 for and 0 against, with a weighted total of 18.0 and a threshold of 6.2.

Vote record:
@N00D13: 1
@PlasmaPower: 1
@cjoh: 1
@datitisev: 1
@eukaryote31: 1
@hongaar: 1
@hubert-pomorski: 1
@laqie: 1
@max-wittig: 1
@phil-r: 1
@qgustavor: 1
@rhengles: 1
@rudehn: 1
@shnist: 1
@tarunbatra: 1
@valfirst: 1
@viktorsec: 1
@voronianski: 1

o3LL committed:
Merge branch 'master' into master

Commit:
Merge branch 'master' into dynamic_voting_window

reddraggone9 committed:
Get `vagrant up` working again

With these changes, the vagrant VM boots and provisions successfully for
me. I'm not sure it ever did for anyone before.
- Aside from pip, the packages are the native dependencies needed to get
  the following `pip3 install` to succeed.
- The change in `synced_folder` should have been made when the Vagrant
  and Docker files were moved. This makes the entire repository
  available under `/vagrant` in the VM.
- Changed the heredoc delimiter to `SCRIPT` because WTH was it `SHELL2`?
- The whitespace changes are because my editor obeys the project's
  `.editorconfig`.
- Removed the comment since it duplicates the README.

Note that these changes *do not* get Chaos working in the VM.

xyproto commented on cron/poll_read_issue_comments.py:
```diff
@@ -0,0 +1,130 @@
+            if(len(sub_command > 1)):
+                args = sub_command[1:]
+                sub_command = sub_command[0]
+                if(sub_command == "label"):
```
A bit shorter:

```python
if sub_command[0] not in ["label", "remove-label", "assign", "unassign"]:
    log_warning = True
```

rudehn replied:
it is, but you still have to check which command you need to run. These commands would be implemented sometime in the future

xyproto replied:
I think it's better to implement things as they are needed. Scaffolding for the future can easily create unmanagable code bases over time. Someone will have to remember to actually do the implementation sometime in the future, and this does not always happen.

flibustier committed:
Reduction of the extended voting window (and btw create the notion of extended voting window)

flibustier committed:
Merge branch 'dynamic_voting_window' of https://github.com/J0hn-/Chaos into dynamic_voting_window

PlasmaPower committed:
Fix flake8 errors and run flake8 in Travis

eukaryote31 committed:
Merge pull request #1 from chaosbot/master

Rebase

chaosbot committed:
merging PR #286: Website Tweaks

https://github.com/chaosbot/Chaos/pull/286: Website Tweaks

Description:
— Changed the “Chaos” text color to red (more suiting).
— Added a notification telling the user to hover over the logo, it is
hard to tell it is animated otherwise.
— Added a moving background that adds more to the “chaos”. (based on
Canva Moving Background from Codepen and it is a background I also use
for my own website.)

:ok_woman: PR passed with a vote of 17 for and 2 against, with a weighted total of 15.0 and a threshold of 6.1.

Vote record:
@DerTiedemann: 1
@KevinAT: 1
@Lastrik: 1
@Leigende: 1
@MUCHZER: 1
@N00D13: -1
@Smittyvb: 1
@albinekb: 1
@amoffat: 1
@eukaryote31: 1
@g19fanatic: 1
@hongaar: -1
@mauran: 1
@meltahawy: 1
@mwaa: 1
@rhengles: 1
@ricardodani: 1
@rudehn: 1
@xyproto: 1

o3LL committed:
fix merge from master

abrad1212 committed:
Add Raspberry Pi support for docker

eukaryote31 committed:
Merge pull request #2 from chaosbot/master

Rebase

abrad1212 committed:
fixed build scripts

Commit:
Move voters.json to server directory

abrad1212 committed:
Create test.txt

chaosbot committed:
merging PR #298: Get `vagrant up` working again

https://github.com/chaosbot/Chaos/pull/298: Get `vagrant up` working again

Description:
With these changes, the vagrant VM boots and provisions successfully for
me. I'm not sure it ever did for anyone before.
- Aside from pip, the packages are the native dependencies needed to get
  the following `pip3 install` to succeed.
- The change in `synced_folder` should have been when the Vagrant and
  Docker files were moved. This makes the entire repository available
  under `/vagrant` in the VM.
- Changed the heredoc delimiter to `SCRIPT` because WTH was it `SHELL2`?
- The whitespace changes are because my editor obeys the projects
  `.editorconfig`
- Removed the comment since it duplicates the README

Note that these changes *do not* get Chaos working in the VM.

:ok_woman: PR passed with a vote of 7 for and 0 against, with a weighted total of 7.0 and a threshold of 6.1.

Vote record:
@amoffat: 1
@eamanu: 1
@eukaryote31: 1
@reddraggone9: 1
@rhengles: 1
@rudehn: 1
@xyproto: 1

abrad1212 committed:
Removed Windows scripts thanks to @reddraggon9

abrad1212 committed:
Removed Windows scripts thanks to @reddraggone9

abrad1212 committed:
I wanna die

abrad1212 committed:
Merge branch 'master' of https://github.com/abrad1212/Chaos

abrad1212 committed:
Moved rpi Docker files to /dev/docker

abrad1212 committed:
Merged branch

chaosbot committed:
merging PR #301: Fix: Merge from master removing website background

https://github.com/chaosbot/Chaos/pull/301: Fix: Merge from master removing website background

Description:
Fix this : https://github.com/chaosbot/Chaos/commit/c0ff053b393f0c604e48269501999f507f8252c8#commitcomment-22303400

From this PR : https://github.com/chaosbot/Chaos/pull/286

🎆 

:ok_woman: PR passed with a vote of 7 for and 0 against, with a weighted total of 7.0 and a threshold of 6.2.

Vote record:
@MUCHZER: 1
@PlasmaPower: 1
@amoffat: 1
@eukaryote31: 1
@meltahawy: 1
@reddraggone9: 1
@rhengles: 1

chaosbot committed:
merging PR #303: Move voters.json to server directory

https://github.com/chaosbot/Chaos/pull/303: Move voters.json to server directory

Description:
This way, we can view the json in-browser

:ok_woman: PR passed with a vote of 9 for and 0 against, with a weighted total of 9.0 and a threshold of 6.2.

Vote record:
@EtherTyper: 1
@PlasmaPower: 1
@abrad1212: 1
@amoffat: 1
@eukaryote31: 1
@ike709: 1
@qgustavor: 1
@reddraggone9: 1
@rhengles: 1

chaosbot committed:
merging PR #300: Fix flake8 errors and run flake8 in Travis

https://github.com/chaosbot/Chaos/pull/300: Fix flake8 errors and run flake8 in Travis

Description:
To fully work, this will require chaosbot's owner @amoffat to create a Travis account and enable it for this repo. Currently I don't think Travis is required for merging, I'll fix that in a later PR. That means that if I did something wrong with Travis, it won't break the repo (hopefully :P).

:ok_woman: PR passed with a vote of 9 for and 0 against, with a weighted total of 9.0 and a threshold of 6.2.

Vote record:
@PlasmaPower: 1
@albinekb: 1
@amoffat: 1
@eukaryote31: 1
@ike709: 1
@reddraggone9: 1
@rhengles: 1
@rudehn: 1
@xyproto: 1

Alexays committed:
Some change to website.

aslafy-z commented on cron/poll_read_issue_comments.py:
```diff
@@ -0,0 +1,130 @@
+    orig_command = command[:]
+    # Check for correct command syntax, ie, subcommands
+    log_warning = False
+    if(len(command)):
```
Please, write pythonic code (remove the parentheses around the ifs) !

rudehn replied:
Ahh my bad!! I totally agree.

chaosbot committed:
merging PR #305: Add Raspberry Pi support for Docker & Docker file fixes

https://github.com/chaosbot/Chaos/pull/305: Add Raspberry Pi support for Docker & Docker file fixes

Description:
Taking what very little I know about **Docker**, I managed to add the needed components for **Raspberry Pi**

It took me about around 3 hours/3 _Original_ Netflix movies to do this :)

:ok_woman: PR passed with a vote of 7 for and 0 against, with a weighted total             of 7.0 and a threshold of 6.1.

Vote record:
@abrad1212: 1
@amoffat: 1
@anythingbot: 1
@eukaryote31: 1
@ike709: 1
@reddraggone9: 1
@rhengles: 1

Commit:
Merge branch 'master' into dynamic_voting_window

Commit:
Merge branch 'master' into dynamic_voting_window

flibustier committed:
Merge remote-tracking branch 'origin/dynamic_voting_window' into dynamic_voting_window

chaosbot committed:
merging PR #308: Some change to website.

https://github.com/chaosbot/Chaos/pull/308: Some change to website.

Description:
- Remove console.log.
- Background in fixed instead of static!
- Vertically centered box.
- Make canvas background color white.
- Move style tag in the css file.
![2017-05-27-093110_1920x1080_scrot](https://cloud.githubusercontent.com/assets/13947260/26519333/b301d918-42bf-11e7-9e1f-47ea5c484bc1.png)


:ok_woman: PR passed with a vote of 8 for and 0 against, with a weighted total             of 8.0 and a threshold of 6.1.

Vote record:
@Its-its: 1
@Leigende: 1
@MCSH: 1
@amoffat: 1
@droogmic: 1
@numaru: 1
@ozyx: 1
@xyproto: 1

chaosbot committed:
merging PR #309: Dynamic voting window 🚀

https://github.com/chaosbot/Chaos/pull/309: Dynamic voting window 🚀

Description:
This PR bounces on suggestions made in #275

That is to say :
- The voting window is now the same as usual (this is now call as "initial_voting_window")
- Two reasons can now extend the voting window (call as "extended_voting_window") :
«  There are two scenarios that the voting window is needed to be long; where it is being hotly debated, and where it gets no votes for many hours. »
And what is the value of the "extending_voting_window" ?
The value depends of the time since the creation date of the repo, following this rule :

![save 2](https://cloud.githubusercontent.com/assets/9994935/26520642/ed492c04-42d6-11e7-9911-704805a4ab8a.png)


When the repository is less than 3 days, the extending voting windows is 3 hours
When the repository is more stable and advanced, 16 days or more, it's 9 hours

:ok_woman: PR passed with a vote of 11 for and 0 against, with a weighted total             of 11.0 and a threshold of 6.2.

Vote record:
@Ealhad: 1
@J0hn-: 1
@Leigende: 1
@akilegaspi: 1
@aslafy-z: 1
@eamanu: 1
@jakejdavis: 1
@paris-ci: 1
@rhengles: 1
@rudehn: 1
@tarunbatra: 1

jakejdavis committed:
Speed up chaos canvas on website

eukaryote31 committed:
Make a voters viewer

phil-r committed:
Add first tests

eukaryote31 committed:
Various formatting things

eukaryote31 committed:
Remove jquery

PlasmaPower committed:
Fix dynamic voting window

PlasmaPower committed:
Close PRs where the backing repo is deleted

Previously, this would cause an insta-merge/insta-reject.

PlasmaPower committed:
Gitignore server/pubkey.txt

chaosbot committed:
Merge remote-tracking branch 'PlasmaPower/fix-dynamic-window'

chaosbot committed:
merging PR #313: Make a voters viewer

https://github.com/chaosbot/Chaos/pull/313: Make a voters viewer

Description:
Now we can see how much each person has voted, as per the voters.json. Not pretty style-wise, but it works. 

:ok_woman: PR passed with a vote of 11 for and 0 against, with a weighted total             of 11.0 and a threshold of 6.2.

Vote record:
@Equim-chan: 1
@J0hn-: 1
@Leigende: 1
@MUCHZER: 1
@PlasmaPower: 1
@TheInitializer: 1
@abrad1212: 1
@davidak: 1
@eukaryote31: 1
@ike709: 1
@phil-r: 1

jakejdavis committed:
Use requestAnimationFrame in chaos canvas on website

chaosbot committed:
merging PR #315: Add first tests

https://github.com/chaosbot/Chaos/pull/315: Add first tests

Description:
This PR:
- adds first unit tests (for 2 functions in `github_api.voting`)
- enables tests on travis
- adds caching to travis (speeds up things 2x)
- if `github_pat.secret` can't be found sets `GITHUB_SECRET` to `None`. The reason behind this is that it's really hard to mock `settings.py` and right now missing `github_pat.secret` file causes exception, therefore tests can't run.

P.S. I've never wrote proper unit tests for python code before, but I hope this will help others to start writing tests for other parts of the codebase

:ok_woman: PR passed with a vote of 8 for and 0 against, with a weighted total             of 8.0 and a threshold of 6.2.

Vote record:
@J0hn-: 1
@MUCHZER: 1
@PlasmaPower: 1
@amoffat: 1
@davidak: 1
@ike709: 1
@mark-i-m: 1
@phil-r: 1

chaosbot committed:
merging PR #316: Close PRs where the backing repo is deleted

https://github.com/chaosbot/Chaos/pull/316: Close PRs where the backing repo is deleted

Description:
Previously, deleting the backing repo caused an insta-merge with #239

:ok_woman: PR passed with a vote of 9 for and 0 against, with a weighted total             of 9.0 and a threshold of 6.2.

Vote record:
@Leigende: 1
@MUCHZER: 1
@PlasmaPower: 1
@aslafy-z: 1
@davidak: 1
@ike709: 1
@loks0n: 1
@nightpool: 1
@phil-r: 1

chaosbot committed:
merging PR #317: Gitignore server/pubkey.txt

https://github.com/chaosbot/Chaos/pull/317: Gitignore server/pubkey.txt

Description:
Auto-generated and should not be committed.

:ok_woman: PR passed with a vote of 8 for and 0 against, with a weighted total             of 8.0 and a threshold of 6.2.

Vote record:
@EtherTyper: 1
@J0hn-: 1
@Leigende: 1
@MUCHZER: 1
@PlasmaPower: 1
@loks0n: 1
@phil-r: 1
@xfix: 1

chaosbot committed:
merging PR #320: Use requestAnimationFrame in chaos canvas on website

https://github.com/chaosbot/Chaos/pull/320: Use requestAnimationFrame in chaos canvas on website

Description:
As recommended in #312 I have used requestAnimationFrame instead of setInterval which speeds up the canvas 

:ok_woman: PR passed with a vote of 8 for and 0 against, with a weighted total             of 8.0 and a threshold of 6.1.

Vote record:
@Leigende: 1
@Smittyvb: 1
@davidak: 1
@eukaryote31: 1
@jakejdavis: 1
@mark-i-m: 1
@phil-r: 1
@xfix: 1

eukaryote31 committed:
Make @smittyvb votes count less

chaosbot committed:
merging PR #323: Make @smittyvb votes count less

https://github.com/chaosbot/Chaos/pull/323: Make @smittyvb votes count less

Description:
Don't worry, this is just a joke. 

:ok_woman: PR passed with a vote of 8 for and 1 against, with a weighted total             of 7.0 and a threshold of 6.2.

Vote record:
@Smittyvb: -1
@Superman132: 1
@eukaryote31: 1
@mark-i-m: 1
@michaelmoussa: 1
@phil-r: 1
@qgustavor: 1
@reddraggone9: 1
@rhengles: 1

rudehn committed:
Merge branch 'patch-3' of https://github.com/rudehn/chaos into homepage_fix

rudehn committed:
fix if statements

rudehn committed:
complaining about imports

rudehn committed:
voting working now

rudehn committed:
add voting window back

aslafy-z commented on cron/poll_read_issue_comments.py:
```diff
@@ -0,0 +1,151 @@
+        else:
+            # Other commands have an = in them
+            sub_command = sub_command.split("=")
+            if len(sub_command > 1):
```
Should be `len(sub_command)  > 1:` right ?

rudehn committed:
Bug fix

rudehn committed:
make travis happy

rudehn committed:
appease travis with code sacrifices

phil-r reviewed (changes requested):
@rudehn I'm forced to downvote, until the mess(110 commits and 44 files changed) is fixed 👎
